dev/com.ibm.ws.concurrent/src/com/ibm/ws/concurrent/ext/ManagedExecutorExtension.java

```diff
@@ -0,0 +1,178 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.concurrent.ext;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Supplier;
+
+import javax.enterprise.concurrent.ManagedExecutorService;
+
+import org.eclipse.microprofile.context.ManagedExecutor;
+import org.eclipse.microprofile.context.ThreadContext;
+
+import com.ibm.websphere.ras.annotation.Trivial;
+import com.ibm.ws.concurrent.WSManagedExecutorService;
+import com.ibm.ws.threading.CompletionStageExecutor;
+import com.ibm.ws.threading.PolicyExecutor;
+import com.ibm.wsspi.resource.ResourceInfo;
+import com.ibm.wsspi.threadcontext.WSContextService;
+
+/**
+ * Extend this interface to intercept and replace resource reference lookups for
+ * <code>managedExecutorService</code>.
+ *
+ * At most one extension implementation can be supplied across the entire system.
+ * A feature that provides this extension point makes itself incompatible
+ * with every other feature that also provides this extension point.
+ * Do not implement if this restriction is unacceptable.
+ */
+@Trivial
+public class ManagedExecutorExtension implements CompletionStageExecutor, ManagedExecutor, ManagedExecutorService, WSManagedExecutorService {
+    private final WSManagedExecutorService executor;
+
+    protected ManagedExecutorExtension(WSManagedExecutorService executor, ResourceInfo resourceInfo) {
```

Comment:
ResourceInfo cannot be removed because it includes information about the resource reference lookup that will be needed by the extension point. One example is that CommonJ has some requirements around the sharing attribute.